features/step_definitions/stepdefs.js

```diff
@@ -615,6 +615,17 @@ When(
             }
         }
 
+        // If this is a request to delete a post, we need to replace the object
+        // name with the object ID as we don't have a way to know the object
+        // ID ahead of time
+        if (requestMethod === 'delete' && path.includes('/post/')) {
+            const objectName = path.split('/').pop();
+            const id = this.activities[objectName].object.id;
+            if (id) {
+                requestPath = path.replace(objectName, encodeURIComponent(id));
+            }
+        }
```

Comment:
Couldn't we tweak the existing logic to handle this also? i.e if `path` is `/thread` OR (`path` is `/post` and req method is `delete`)?

I think the only difference is your referencing an activity rather than an object? (think we might already have a pattern for handling this)